Tasks/XamariniOSV2/xamarinios.ts

```diff
@@ -28,6 +29,11 @@ async function run() {
     try {
         tl.setResourcePath(path.join(__dirname, 'task.json'));
 
+        // Check platform is macOS since demands are not evaluated on Hosted pools
+        if (!/^darwin/.test(os.platform())) {
+            throw tl.loc('BuildRequiresMac');
```

Comment:
An improvement is `throw new Error(tl.loc('BuildRequiresMac'));` It will preserve the stack trace correctly.